### PR TITLE
Add support for Avro Nullable annotation

### PR DESCRIPTION
**Pull request by HoelzelJon**
A project I work on makes frequent use of the apache avro Nullable annotation (`org.apache.avro.reflect.Nullable`). Adding support for the avro nullable annotation would significantly reduce the number of false positive warnings generated by spotbugs.

----

Make sure these boxes are checked before submitting your PR -- thank you!

- [x] Added an entry into `CHANGELOG.md` if you have changed SpotBugs code

### CODE REVIEW

**Comment by HoelzelJon** [comment]
I am unable to run the full build as described in the contribution guidelines (see https://github.com/spotbugs/spotbugs/issues/2065), but everything seems to work fine when I skip the eclipsePlugin tasks.

**Review by KengoTODA — APPROVED** [review]

File: spotbugsTestCases/src/fakeAnnotations/org/apache/avro/reflect/Nullable.java [hunk 1]
```diff
@@ -0,0 +1,35 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
```
Comment (ThrawnCA) [inline spotbugsTestCases/src/fakeAnnotations/org/apache/avro/reflect/Nullable.java:2]: SpotBugs doesn't belong to Apache. Why is this copyright notice in use?

Comment (HoelzelJon) [reply spotbugsTestCases/src/fakeAnnotations/org/apache/avro/reflect/Nullable.java:2]: Sorry for the slow response

I copied the file from the Apache avro source, following what seems to have been done for the `android.support.annotation` classes (which use the same license).

I'm no expert on how this license works; if you think it's more reasonable to remove the license block here, I'd be happy to do so.

Comment (ThrawnCA) [reply spotbugsTestCases/src/fakeAnnotations/org/apache/avro/reflect/Nullable.java:2]: Ah. That does make sense.

**Comment by JuditKnoll** [comment]
It seems like during the merge, this feature in the CHANGELOG.md got to the wrong place, not to the unreleased section, but into 4.7.2.

**Comment by hazendaz** [comment]
Thank you will get that fixed.  We need a better way on change logs.  Anything aging is doomed to merge conflict or get placed incorrectly.

Sent from my Verizon, Samsung Galaxy smartphone
Get Outlook for Android<https://aka.ms/AAb9ysg>
________________________________

**Pull request merged**